x/virtualgroup/types/params.pb.go

```diff
@@ -0,0 +1,528 @@
+// Code generated by protoc-gen-gogo. DO NOT EDIT.
+// source: greenfield/virtualgroup/params.proto
+
+package types
+
+import (
+	fmt "fmt"
+	_ "github.com/cosmos/cosmos-proto"
+	github_com_cosmos_cosmos_sdk_types "github.com/cosmos/cosmos-sdk/types"
+	_ "github.com/cosmos/gogoproto/gogoproto"
+	proto "github.com/cosmos/gogoproto/proto"
+	io "io"
+	math "math"
+	math_bits "math/bits"
+)
+
+// Reference imports to suppress errors if they are not otherwise used.
+var _ = proto.Marshal
+var _ = fmt.Errorf
+var _ = math.Inf
+
+// This is a compile-time assertion to ensure that this generated file
+// is compatible with the proto package it is being compiled against.
+// A compilation error at this line likely means your copy of the
+// proto package needs to be updated.
+const _ = proto.GoGoProtoPackageIsVersion3 // please upgrade the proto package
+
+// Params defines the parameters for the module.
+type Params struct {
+	// deposit_denom defines the staking coin denomination.
+	DepositDenom string `protobuf:"bytes,1,opt,name=deposit_denom,json=depositDenom,proto3" json:"deposit_denom,omitempty"`
+	// store price, in bnb wei per charge byte
+	GvgStakingPrice github_com_cosmos_cosmos_sdk_types.Dec `protobuf:"bytes,2,opt,name=gvg_staking_price,json=gvgStakingPrice,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Dec" json:"gvg_staking_price"`
```

Comment:
If we change `GvgStakingPrice` or `DepositDenom`, big problem will occur at current implementation, any method to protect it?

Reply:
We can add some protest in `UpdateParams` message transaction.

Reply:
yes, please do this.